velox/external/duckdb/CMakeLists.txt

```diff
@@ -14,13 +14,10 @@ add_subdirectory(tpch)
 add_compile_definitions(DISABLE_DUCKDB_REMOTE_INSTALL)
 add_compile_definitions(BUILD_PARQUET_EXTENSION)
 
-# Certain Clang compiler versions throw "deprecated declarations" warnings for
+# Certain compiler versions throw "deprecated declarations" warnings for
 # DuckDB's parquet-amalgamation.h. This warning is from the Thrift library
```

Comment:
As DuckDB is vendored anyway, can we not patch that file to disable the warning instead of disabling this warning in the entire code base?

Reply:
We have a semi-painless process to update these files (which we go through often), so requiring patches to the cpp files  complicate things. Agreed that upstreaming these changes to DuckDB itself would be great, but I'm not sure how receptive they would be, considering they have much higher portability standards than we do.

Reply:
Hm, ok makes sense for now. IIRC you vendor ddb because you use internal apis right?

Reply:
Ok I checked the vendored ddb and this code is mirrored there: https://github.com/facebookincubator/velox/blob/main/velox/external/duckdb/CMakeLists.txt#L20-L32

As the duckdb target is created it should be added via `target_compile_options(duckdb ...)` this will add this flag only to the duckdb target and not to the entire code base.